### PR TITLE
ThreatQ Integration 1.0.6

### PR DESCRIPTION
**Pull request by robert-phe**
robert.asztalos@3pillarglobal.com

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Description
- Removed threatq-search-by-name command
- Fix default parameters for the following commands:
    - threatq-search-by-name
    - threatq-upload-file
    - threatq-get-all-indicators
    - threatq-get-all-events
    - threatq-get-all-adversaries

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/ThreatQ/Integrations/ThreatQ_v2/ThreatQ_v2.py [hunk 1]
```diff
@@ -1186,8 +1103,8 @@ def download_file_command():
 
 def get_all_objs_command(obj_type):
     args = demisto.args()
-    page = int(args.get('page'))
-    limit = int(args.get('limit'))
+    page = int(args.get('page')) or 0
+    limit = int(args.get('limit')) or 50
```
Comment (reutshal) [inline Packs/ThreatQ/Integrations/ThreatQ_v2/ThreatQ_v2.py:1107]:
```suggestion
    page = int(args.get('page', 0))
    limit = int(args.get('limit'), 50)
```

File: Packs/ThreatQ/Integrations/ThreatQ_v2/ThreatQ_v2.py [hunk 2]
```diff
@@ -1127,7 +1044,7 @@ def upload_file_command():
     args = demisto.args()
     entry_id = args.get('entry_id')
     title = args.get('title')
-    malware_safety_lock = args.get('malware_safety_lock')
+    malware_safety_lock = args.get('malware_safety_lock') or 'off'
```
Comment (reutshal) [inline Packs/ThreatQ/Integrations/ThreatQ_v2/ThreatQ_v2.py:1047]:
```suggestion
    malware_safety_lock = args.get('malware_safety_lock', 'off')
```

File: Packs/ThreatQ/Integrations/ThreatQ_v2/ThreatQ_v2.py [hunk 3]
```diff
@@ -656,74 +637,10 @@ def aggregate_search_results(indicators, default_indicator_type, generic_context
...
-    )
-
-    return_outputs(readable, entry_context, search_results)
-
-
 def search_by_name_command():
     args = demisto.args()
     name = args.get('name')
-    limit = args.get('limit')
+    limit = args.get('limit') or '10'
```
Comment (reutshal) [inline Packs/ThreatQ/Integrations/ThreatQ_v2/ThreatQ_v2.py:643]:
```suggestion
    limit = args.get('limit', '10')
```

**Review by reutshal — COMMENTED** [review]
Hi, thanks for contributing this!
Overall looks good, few things - 
* I am not clear why did you remove the command?
This could cause BC breaking for customers using this.
* I'm suggesting adding the default values to the arguments in the yml files as well, that way the customers will be aware of the default values.

File: Packs/ThreatQ/Integrations/ThreatQ_v2/ThreatQ_v2.py (hunk 3 above)
Comment (robert-phe) [reply Packs/ThreatQ/Integrations/ThreatQ_v2/ThreatQ_v2.py]: 2bd88f9

File: Packs/ThreatQ/Integrations/ThreatQ_v2/ThreatQ_v2.py (hunk 2 above)
Comment (robert-phe) [reply Packs/ThreatQ/Integrations/ThreatQ_v2/ThreatQ_v2.py]: 2bd88f9

File: Packs/ThreatQ/Integrations/ThreatQ_v2/ThreatQ_v2.py (hunk 1 above)
Comment (robert-phe) [reply Packs/ThreatQ/Integrations/ThreatQ_v2/ThreatQ_v2.py]: 2bd88f9

**Comment by robert-phe** [comment]
Hello,

- The command was removed because it uses the TQ deprecated endpoint "/search/advanced".
- As far as I saw the default values are added to the yaml and displayed in the UI, but it seems that the args.get(<param>) returns None even if the default is set in the yaml

**Comment by reutshal** [comment]
> Hello,
> 
> * The command was removed because it uses the TQ deprecated endpoint "/search/advanced".
> * As far as I saw the default values are added to the yaml and displayed in the UI, but it seems that the args.get(-param-) returns None even if the default is set in the yaml

In that case, sounds like the command should be deprecated rather than removed from the integration.
You can easily do that by adding: `deprecated: true` under this command in the yml file and then customers can still use it but they will know it is deprecated and will not be supported.

**Comment by robert-phe** [comment]
As discussed, re-added the code for `threatq-advanced-search` and added the `deprecated: true` in the yaml file.
Note: The `threatq-advanced-search` is still removed from the readme file.
9badb2e

**Review by reutshal — APPROVED** [review]

**Pull request merged**